internal/resource/tree.go

```diff
@@ -92,11 +92,17 @@ func (b *treeBuilder) Build() *tree {
 	for _, idx := range b.byRef {
 		t.byManiRef[idx.Resource.ManifestRef] = idx
 
-		// CRs are dependent on their CRDs
+		// CRs are dependent on their CRDs, except in delete where its reversed
+		// can this introduce circular dependencies if readines/deltion groups say opposite?
 		crd, ok := b.byDefiningGK[idx.Resource.GVK.GroupKind()]
 		if ok {
-			idx.PendingDependencies[crd.Resource.Ref] = struct{}{}
-			crd.Dependents[idx.Resource.Ref] = idx
+			if idx.Resource.compositionDeleted {
```

Comment:
any reason to invert cr/crd ordering rather than ignoring it during deletion?

Reply:
I assume trying to delete a CRD while CR's still exist will just fail so why not wait? You could say the same thing about creation. Eventually the crd will be applied and then the CR will go so either we should ignore for both or obey ordering for both?

Reply:
Comment updated and UT added. Feel free to make suggestions to comment.